test/utility.py

```diff
@@ -98,3 +100,65 @@ def get_functions_with_prefix(prefix, obj):
     :return list of function objects
     """
     return [getattr(obj, func) for func in dir(obj) if func.startswith(prefix)]
+
+
+class _RedirectStream:
+    """
+    Context manager for temporarily redirecting a stream to another file.
+    Copied from py35 contextlib because _RedirectStream is not available in py34 contextlib.
+    """
+    _stream = None
+
+    def __init__(self, new_target):
+        self._new_target = new_target
+        # We use a list of old targets to make this CM re-entrant
+        self._old_targets = []
+
+    def __enter__(self):
+        self._old_targets.append(getattr(sys, self._stream))
+        setattr(sys, self._stream, self._new_target)
+        return self._new_target
+
+    def __exit__(self, exctype, excinst, exctb):
+        setattr(sys, self._stream, self._old_targets.pop())
+
+
+class RedirectStdout(_RedirectStream):
+    """
+    Context manager for temporarily redirecting stdout to another file.
+    Copied from py35 contextlib because RedirectStdout is not available in py34 contextlib.
+    """
+
+    _stream = "stdout"
+
+
+class RedirectStderr(_RedirectStream):
+    """
+    Context manager for temporarily redirecting stderr to another file.
+    Copied from py35 contextlib because RedirectStderr is not available in py34 contextlib.
+    """
+
+    _stream = "stderr"
+
+
+def mute(function):
+    """
+    This function provides an function decorator which redirects the stderr and stdout stream to an stream object
+    in order to prevent console output.
+    """
+    def _mute(*args, **kwargs):
+        """
+        This method absorbs stderr and stdout streams.
+        :param args: extra positional generic arguments
+        :param kwargs: extra keyword arguments generic argument
+        :return: return value of `function`
+        """
+        output_sink = StringIO()
+        try:
+            # with redirect_stdout(output_sink):
+            with RedirectStderr(output_sink), RedirectStdout(output_sink):
```

Comment:
could you here check if the "original" `Redirect...` classes are available, and if so, use them? That way we do not block possible updates/bug fixes in future python versions.

Reply:
I fixed that in
https://github.com/nils-wisiol/pypuf/blob/995e4578e02f371c2185b9c0c2f9db246ae85c14/test/utility.py#L157-L166 .